### PR TITLE
Support typescript mode

### PR DESCRIPTION
**Pull request by nyrosmith**
Associated Issue: #5498

### CODE REVIEW

**Comment by jasonLaster** [comment]
This is great. Small eslint issue, feel free bumping the complexity

**Comment by nyrosmith** [comment]
I'd rather refactor the function. I hope it becomes more readable / maintainable...

**Comment by jasonLaster** [comment]
that works for me too

**Comment by nyrosmith** [comment]
@jasonLaster Ok, I increased the complexity rule because I could not find a way to decrease complixity in `getMode` which makes it more readable....

File: package.json [hunk 1]
```diff
@@ -18,40 +18,31 @@
     "start": "node bin/dev-server",
     "start-app": "TARGET=application node bin/dev-server",
     "flow": "flow",
-    "eslint-check":
-      "eslint --print-config .eslintrc.js | eslint-config-prettier-check",
+    "eslint-check": "eslint --print-config .eslintrc.js | eslint-config-prettier-check",
```
Comment (jasonLaster) [inline package.json]: was this package.json formatting intentional? I think prettier could help here

**Pull request merged**